presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java

```diff
@@ -0,0 +1,254 @@
+    }
+
+    public static long getFileSize(Page statisticsPage, int position)
+    {
+        // FileStatistics page layout:
+        //
+        // fileSize   rowCount
+        //  X             X
+
+        if (position < 0 || position >= statisticsPage.getPositionCount()) {
+            throw new PrestoException(MALFORMED_HIVE_FILE_STATISTICS, format("Invalid position: %d specified for FileStatistics page", position));
+        }
+        return BIGINT.getLong(statisticsPage.getBlock(FILE_SIZE_CHANNEL), position);
+    }
+
+    public static Optional<Page> createPartitionManifest(PartitionUpdate partitionUpdate)
+    {
+        // Manifest Page layout:
+        //   fileName    fileSize
+        //      X           X
+        //      X           X
+        //      X           X
+        // ....
+        PageBuilder manifestBuilder = new PageBuilder(ImmutableList.of(VARCHAR, BIGINT));
+        BlockBuilder fileNameBuilder = manifestBuilder.getBlockBuilder(0);
+        BlockBuilder fileSizeBuilder = manifestBuilder.getBlockBuilder(1);
+        for (FileWriteInfo fileWriteInfo : partitionUpdate.getFileWriteInfos()) {
+            if (!fileWriteInfo.getFileSize().isPresent()) {
+                return Optional.empty();
+            }
+            manifestBuilder.declarePosition();
+            VARCHAR.writeSlice(fileNameBuilder, utf8Slice(fileWriteInfo.getWriteFileName()));
+            BIGINT.writeLong(fileSizeBuilder, fileWriteInfo.getFileSize().get());
+        }
+        return Optional.of(manifestBuilder.build());
+    }
+
+    public static Map<String, String> updatePartitionMetadataWithFileNamesAndSizes(PartitionUpdate partitionUpdate, Map<String, String> metadata)
+    {
+        ImmutableMap.Builder<String, String> partitionMetadata = ImmutableMap.builder();
+        List<FileWriteInfo> fileWriteInfos = new ArrayList<>(partitionUpdate.getFileWriteInfos());
+
+        if (!partitionUpdate.containsNumberedFileNames()) {
+            // Filenames starting with ".tmp.presto" will be renamed in TableFinishOperator. So it doesn't make sense to store the filenames in manifest
+            return metadata;
+        }
+
+        // Sort the file infos based on fileName
+        fileWriteInfos.sort(Comparator.comparing(info -> Integer.valueOf(info.getWriteFileName())));
+
+        List<String> fileNames = fileWriteInfos.stream().map(FileWriteInfo::getWriteFileName).collect(toImmutableList());
+        List<Long> fileSizes = fileWriteInfos.stream().map(FileWriteInfo::getFileSize).filter(Optional::isPresent).map(Optional::get).collect(toImmutableList());
+
+        if (fileSizes.size() < fileNames.size()) {
+            if (fileSizes.isEmpty()) {
+                // These files may not have been written by OrcFileWriter. So file sizes not available.
+                return metadata;
+            }
+            throw new PrestoException(
+                    MALFORMED_HIVE_FILE_STATISTICS,
+                    format(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;size&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 136. Consider formatting this data with &#39;%d&#39; instead.